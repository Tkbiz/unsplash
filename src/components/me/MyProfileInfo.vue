<template>
  <section class="container-fluid profile-info">
    <v-card>
      <v-tabs
          v-model="tab"
          icons-and-text
          show-arrows>
        <v-tabs-slider></v-tabs-slider>
        <v-tab href="#tab-1">
          Photos
          <v-icon>mdi-image-multiple</v-icon>
        </v-tab>
        <v-tab href="#tab-2">
          Likes
          <v-icon>mdi-heart</v-icon>
        </v-tab>
        <v-tab href="#tab-3">
          Collections
          <v-icon>mdi-layers-triple</v-icon>
        </v-tab>
        <v-tab href="#tab-4">
          Stats
          <v-icon>mdi-chart-bar</v-icon>
        </v-tab>
      </v-tabs>
      <v-tabs-items v-model="tab">
        <v-tab-item
            v-for="i in 4"
            :key="i"
            :value="'tab-' + i">
          <v-card
              v-if="i === 1"
              flat>
            <div v-if="logInUser.photos.length > 0">
              <v-card-text
                  v-if="logInUser.photos">
                <v-col
                    cols="12"
                    sm="6"
                    md="4"
                    lg="3"
                    v-for="photo in logInUser.photos"
                    :key="photo.id">
                  <img
                      @click.prevent="downloadPhoto(photo.urls.small)"
                      :src="photo.urls.small ? photo.urls.small : '@/assets/images/no-image.png'"
                      class="profile-img cursor-pointer"
                      alt="user photo">
                </v-col>
              </v-card-text>
            </div>
            <div v-else>
              <v-card-text>
                <v-col cols="12" class="collection-title">
                  {{ noPhotos }}
                </v-col>
              </v-card-text>
            </div>
          </v-card>
          <v-card
              v-else-if="i === 2"
              flat>
            <div v-if="userLikedPhotos.length > 0">
              <v-card-text>
                <v-col
                    v-if="userLikedPhotos"
                    cols="12"
                    sm="6"
                    md="4"
                    lg="3"
                    v-for="photo in userLikedPhotos"
                    :key="photo.id">
                  <img
                      @click.prevent="downloadPhoto(photo.urls.small)"
                      :src="photo.urls.small ? photo.urls.small : '@/assets/images/no-image.png'"
                      class="profile-img cursor-pointer"
                      alt="user photo">
                </v-col>
                <v-col
                    v-else
                    cols="12">
                  <img
                      src="@/assets/images/no-photos.png"
                      class="profile-img"
                      alt="no photos">
                </v-col>
              </v-card-text>
            </div>
            <div v-else>
              <v-card-text>
                <v-col cols="12" class="collection-title">
                  {{ noPhotos }}
                </v-col>
              </v-card-text>
            </div>
          </v-card>
          <v-card
              v-else-if="i === 3"
              flat>
            <div v-if="userCollections.length > 0">
              <v-card-text
                  v-for="item of userCollections"
                  :key="item.id">
                <v-col cols="12" class="collection-title">
                  {{ item.title }}
                </v-col>
                <v-col
                    cols="12"
                    sm="6"
                    md="4"
                    lg="3"
                    v-for="photo in item.preview_photos"
                    :key="photo.id">
                  <img
                      @click.prevent="downloadPhoto(photo.urls.small)"
                      :src="photo.urls.small ? photo.urls.small : '@/assets/images/no-image.png'"
                      class="profile-img cursor-pointer"
                      alt="user photo">
                </v-col>
              </v-card-text>
            </div>
            <div v-else>
              <v-card-text>
                <v-col cols="12" class="collection-title">
                  {{ noPhotos }}
                </v-col>
              </v-card-text>
            </div>
          </v-card>
          <v-card
              v-else-if="i === 4"
              flat>
            <v-card-text>
              <div v-if="logInUser.followers_count"
                   class="profile-info">
                Followers:
                {{ logInUser.followers_count }}
              </div>
              <div v-if="logInUser.following_count"
                   class="profile-info">
                Following:
                {{ logInUser.following_count }}
              </div>
              <div v-if="logInUser.total_likes"
                   class="profile-info">
                Likes:
                {{ logInUser.total_likes }}
              </div>
              <div v-if="logInUser.downloads"
                   class="profile-info">
                Downloads:
                {{ logInUser.downloads }}
              </div>
            </v-card-text>
          </v-card>
        </v-tab-item>
      </v-tabs-items>
    </v-card>
  </section>
</template>

<script>
export default {
  name: 'MyProfileInfo',
  props: {
    logInUser: {
      type: Object,
      required: true
    }
  },
  data: () => ({
    tab: null,
    noPhotos: 'Not found any photos!'
  }),
  methods: {
    async downloadPhoto(url) {
      await this.$store.dispatch('downloadPhoto', url)
    }
  },
  computed: {
    userLikedPhotos() {
      return this.$store.getters.userLikedPhotos
    },
    userCollections() {
      return this.$store.getters.userCollections
    }
  },
  async mounted() {
    await this.$store.dispatch('fetchUserLikedPhotos', this.logInUser.username)
    await this.$store.dispatch('fetchUserCollections', this.logInUser.username)
  }
}
</script>
