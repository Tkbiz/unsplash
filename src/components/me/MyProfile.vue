<template>
  <div>
    <section class="container profile">
      <v-row>
        <v-col cols="12"
               class="py-0 mb-9">
          <v-row
              class="mb-7"
              align="start"
              justify="center">
            <v-avatar
                v-if="logInUser.profile_image"
                size="128">
              <img
                  :src="logInUser.profile_image.large ? logInUser.profile_image.large : '@/assets/images/no-image.png'"
                  alt="John">
            </v-avatar>
            <div>
              <h2 v-if="logInUser.name"
                  class="profile-title">
                {{ logInUser.name }}
              </h2>
              <div v-if="logInUser.location"
                   class="profile-info">
                {{ logInUser.location }}
              </div>
              <v-list-item
                  v-if="twitterUrl"
                  :href="twitterUrl"
                  class="profile-info"
                  target="_blank">
                {{ twitterUrl }}
              </v-list-item>
              <v-list-item
                  v-if="instagramUrl"
                  :href="instagramUrl"
                  class="profile-info"
                  target="_blank">
                {{ instagramUrl }}
              </v-list-item>
            </div>
          </v-row>
          <div v-if="logInUser.bio"
               class="col-12 col-md-9 col-lg-6 profile-description">
            {{ logInUser.bio }}
          </div>
        </v-col>
      </v-row>
    </section>
    <MyProfileInfo
        :logInUser="logInUser"/>
  </div>
</template>

<script>
import MyProfileInfo from '@/components/me/MyProfileInfo'

export default {
  name: 'MyProfile',
  props: {
    logInUser: {
      type: Object,
      required: true
    }
  },
  components: {
    MyProfileInfo
  },
  computed: {
    twitterUrl() {
      if (this.logInUser.twitter_username) {
        return process.env.VUE_APP_UNSPLASH_TWITTER_URL + this.logInUser.twitter_username
      }
    },
    instagramUrl() {
      if (this.logInUser.instagram_username) {
        return process.env.VUE_APP_UNSPLASH_INSTAGRAM_URL + this.logInUser.instagram_username
      }
    }
  }
}
</script>
