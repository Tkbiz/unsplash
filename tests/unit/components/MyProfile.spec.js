import {mount, createLocalVue} from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import MyProfile from '@/components/me/MyProfile'
import MyProfileInfo from '@/components/me/MyProfileInfo'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

describe('MyProfile.vue', () => {
  let vuetify
  let wrapper
  let actions
  let getters
  let store

  beforeEach(() => {
    vuetify = new Vuetify()

    getters = {
      userLikedPhotos: () => ([]),
      userCollections: () => ([])
    }
    actions = {
      fetchUserLikedPhotos: jest.fn(),
      fetchUserCollections: jest.fn()
    }
    store = new Vuex.Store({
      getters,
      actions
    })

    wrapper = mount(MyProfile, {
      localVue,
      vuetify,
      store,
      propsData: {
        logInUser: {
          photos: []
        }
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it("renders MyProfileInfo", () => {
    expect(wrapper.exists(MyProfileInfo)).toBe(true)
  })

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders props when passed', () => {
    const logInUser = {
      photos: []
    }
    expect(wrapper.props()).toEqual({logInUser})
  })
})
