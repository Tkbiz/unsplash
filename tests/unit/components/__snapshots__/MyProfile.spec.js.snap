// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile.vue should match snapshot 1`] = `
<div>
  <section class="container profile">
    <div class="row">
      <div class="py-0 mb-9 col col-12">
        <div class="row mb-7 align-start justify-center">
          <!---->
          <div>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
        <!---->
      </div>
    </div>
  </section>
  <section class="container-fluid profile-info">
    <div class="v-card v-sheet theme--light">
      <div class="v-tabs v-tabs--icons-and-text theme--light">
        <div role="tablist" class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile v-tabs-bar--show-arrows primary--text">
          <div class="v-slide-group__prev v-slide-group__prev--disabled">
            <transition-stub name="fade-transition"><i aria-hidden="true" class="v-icon notranslate v-icon--disabled mdi mdi-chevron-left theme--light"></i></transition-stub>
          </div>
          <div class="v-slide-group__wrapper">
            <div class="v-slide-group__content v-tabs-bar__content">
              <div class="v-tabs-slider-wrapper" style="transition: none;">
                <div class="v-tabs-slider"></div>
              </div> <a tabindex="0" href="#tab-1" aria-selected="true" role="tab" class="v-tab v-tab--active">
                Photos
                <i aria-hidden="true" class="v-icon notranslate mdi mdi-image-multiple theme--light"></i></a> <a tabindex="0" href="#tab-2" aria-selected="false" role="tab" class="v-tab">
                Likes
                <i aria-hidden="true" class="v-icon notranslate mdi mdi-heart theme--light"></i></a> <a tabindex="0" href="#tab-3" aria-selected="false" role="tab" class="v-tab">
                Collections
                <i aria-hidden="true" class="v-icon notranslate mdi mdi-layers-triple theme--light"></i></a> <a tabindex="0" href="#tab-4" aria-selected="false" role="tab" class="v-tab">
                Stats
                <i aria-hidden="true" class="v-icon notranslate mdi mdi-chart-bar theme--light"></i></a>
            </div>
          </div>
          <div class="v-slide-group__next v-slide-group__next--disabled">
            <transition-stub name="fade-transition"><i aria-hidden="true" class="v-icon notranslate v-icon--disabled mdi mdi-chevron-right theme--light"></i></transition-stub>
          </div>
        </div>
      </div>
      <div class="v-window v-item-group theme--light v-tabs-items">
        <div class="v-window__container">
          <transition-stub name="">
            <div class="v-window-item v-window-item--active" id="tab-1">
              <div class="v-card v-card--flat v-sheet theme--light">
                <div>
                  <div class="v-card__text">
                    <div class="collection-title col col-12">
                      Not found any photos!
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </transition-stub>
          <transition-stub name="">
            <!---->
          </transition-stub>
          <transition-stub name="">
            <!---->
          </transition-stub>
          <transition-stub name="">
            <!---->
          </transition-stub>
        </div>
      </div>
    </div>
  </section>
</div>
`;
